sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -360,10 +360,13 @@ static bool _pio_from_iface_addr(gnrc_pktsnip_t **res, gnrc_ipv6_netif_addr_t *a
         DEBUG(" - PIO for %s/%" PRIu8 "\n", ipv6_addr_to_str(addr_str, &addr->addr,
                                                              sizeof(addr_str)),
               addr->prefix_len);
-        *res = gnrc_ndp_opt_pi_build(addr->prefix_len, (addr->flags &
-                                     (GNRC_IPV6_NETIF_ADDR_FLAGS_NDP_AUTO |
-                                      GNRC_IPV6_NETIF_ADDR_FLAGS_NDP_ON_LINK)),
-                                     addr->valid, addr->preferred, &addr->addr, next);
+
+        uint8_t flags = addr->flags | GNRC_IPV6_NETIF_ADDR_FLAGS_NDP_AUTO;
```

Comment:
Also shouldn't the auto-flag be set rather for the address and no for every PIO?